compiler/AST/AggregateType.cpp

```diff
@@ -234,6 +237,12 @@ void AggregateType::verify() {
         INT_FATAL(this, "Substitution value not in tree");
     }
   }
+
+  // Should we just change all checks of this flag to a check against
+  // postinit being non-NULL?  Or a nice method query?
+  if ((postinit != NULL) != this->symbol->hasFlag(FLAG_HAS_POSTINIT)) {
```

Comment:
My brain is terrible at boolean operations :D

`this->symbol->hasFlag(FLAG_HAS_POSTINIT) && (postinit == nullptr)` reads closer to English to me. But that's nitpick.

More importantly, I think we want to use `nullptr` as opposed to `NULL`.

Reply:
I think your proposal is a different check (?):  Mine makes sure that postinit is not NULL if-and-only-if it has a postinit flag—that is, it prevents us from accidentally getting into a state where we've set the flag but not the field or vice-versa (and in fact, I had ~60 tests fail due to this inconsistency in my first paratest run).  In contrast, yours checks that it definitely has a postinit (right?), which not all aggregate types will.

Reply:
(I'll make the nullptr change, though… showing my age!)

Reply:
(though in my defense, there are _lots_ of `NULL`s in code surrounding all the code I added…)

Reply:
> I think your proposal is a different check (?): Mine makes sure that postinit is not NULL if-and-only-if it has a postinit flag—that is, it prevents us from accidentally getting into a state where we've set the flag but not the field or vice-versa (and in fact, I had ~60 tests fail due to this inconsistency in my first paratest run). In contrast, yours checks that it definitely has a postinit (right?), which not all aggregate types will.



Ugh.. yeah, you are right. I would probably make it a nested `if`, but it is getting more into a matter of style. You can consider this resolved.